Languages/es/03_Funcion_es/readme.md

```diff
@@ -0,0 +1,172 @@
+
+Comunidad: [Discord](https://discord.gg/5akcruXrsk)｜[Wechat](https://docs.google.com/forms/d/e/1FAIpQLSe4KGT8Sh6sJ7hedQRuIYirOoZK_85miz3dw7vA1-YjodgJ-A/viewform?usp=sf_link)｜[Website wtf.academy](https://wtf.academy)
+
+Los códigos y tutoriales están como código abierto en GitHub: [github.com/AmazingAng/WTFSolidity](https://github.com/AmazingAng/WTFSolidity)
+
+
+---
+
+## Función (Function)
+
+Este es el formato de una función en Solidity:
+
+```solidity
+    function <nombre de la función>(<tipos de los parámetros>) [internal|external] [pure|view|payable] [returns (<tipos de retorno>)]
+```
+
+Puede parecer complejo, pero desglosémoslo pieza por pieza (los corchetes cuadrados indican palabras clave opcionales):
+
+
+1. `function`: Para escribir una función, necesitas comenzar con la palabra clave `function`.
+
+2. `<function name>`: El nombre de la función.
+
+3. `(<parameter types>)`: Los tipos y nombres de los parámetros de entrada.
+
+3. `[internal|external|public|private]`: Especificadores de visibilidad de la función. Especificar explicitamente la visibilidad es una buena práctica, por lo que debes especificarla para cada función. Hay 4 tipos de ellos:
+
+   - `public`: Visible para todos.
+
+   - `private`: Solo puede ser accedido dentro de este contrato, los contratos derivados no pueden usarlo.
+
+   - `external`: Solo puede ser llamado desde otros contratos. Pero también puede ser llamado por `this.f()` dentro del contrato, donde `f` es el nombre de la función.
+
+   - `internal`: Solo puede ser accedido internamente y por contratos que derivan de él.
+
+    **Nota 1**: `public` es la visibilidad predeterminada para las funciones.
+    
+    **Nota 2**: `public|private|internal` también puede ser usado en variables de estado. Las variables públicas generarán automáticamente funciones `getter` para consultar valores.
+    
+    **Note 3**: La visibilidad predeterminada para las variables de estado es `internal`.
+
+4. `[pure|view|payable]`: Son palabras clave que dictan el comportamiento de las funciones de Solidity. `payable` (pagable) is facil de entender. Se puede enviar `ETH` a un contrato con funciones `payable` (pagables). `pure` y `view` se introducen en la próxima sección.
+
+5. `[returns (<tipos de retorno>)]`: Tipos y nombres de las variables de retorno.
+
+## WTF es `Pure` y `View`?
+
+Cuando empecé a aprender `Solidity`, no entendía en absoluto las palabras clave `pure` y `view`, ya que no son comunes en otros lenguajes. `Solidity` agregó estas dos palabras clave debido al costo del gas `(gas fee)`. Las variables de estado del contrato se almacenan en la cadena de bloques, y el costo del gas `(gas fee)` es muy elevado. Si no reescribes estas variables, no necesitas pagar `gas`. No necesitas pagar gas por llamar a funciones `pure` y `view`.
+
+Las siguientes acciones se consideran modificaciones del estado:
+
+1. Escribir en variables de estado.
+
+2. Emitir eventos.
+
+3. Crear otros contratos.
+
+4. Usar `selfdestruct`.
```

Comment:
maybe self.destruct?

Reply:
selfDestruct is a function
![image](https://github.com/scguaquetam/WTF-Solidity/assets/23249805/81253e83-a0ea-418b-9a25-beae02f7235b)
https://www.alchemy.com/overviews/selfdestruct-solidity